pkgs/development/python-modules/datalad-next/default.nix

```diff
@@ -0,0 +1,106 @@
+{
+  annexremote,
+  buildPythonPackage,
+  datalad,
+  datasalad,
+  fetchFromGitHub,
+  git,
+  git-annex,
+  humanize,
+  lib,
+  more-itertools,
+  psutil,
+  pythonOlder,
+  setuptools,
+  openssh,
+  unzip,
+  webdavclient3,
+}:
+
+buildPythonPackage rec {
+  pname = "datalad-next";
+  version = "1.5.0";
+  pyproject = true;
+  disabled = pythonOlder "3.12";
+
+  src = fetchFromGitHub {
+    owner = "datalad";
+    repo = "datalad-next";
+    rev = "refs/tags/${version}";
+    hash = "sha256-fqP6nG2ncDRg48kvlsmPjNBOzfQp9+7wTcGvsYVrRzA=";
+  };
+
+  nativeBuildInputs = [ git ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    annexremote
+    datasalad
+    datalad
+    humanize
+    more-itertools
+  ];
+
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  nativeCheckInputs = [
+    webdavclient3
+    psutil
+    git-annex
+    datalad
+    openssh
+    unzip
+  ];
+
+  disabledTests = [
+    # remotes available after datalad-next install (through `console_scripts`), but not yet in $PATH during test
+    "test_uncurl_addurl_unredirected"
+    "test_uncurl"
+    "test_uncurl_ria_access"
+    "test_uncurl_store"
+    "test_uncurl_remove"
+    "test_uncurl_testremote"
+    "test_replace_add_archive_content"
+    "test_annex_remote"
+    "test_export_remote"
+    "test_annex_remote_autorepush"
+    "test_export_remote_autorepush"
+    "test_typeweb_annex"
+    "test_typeweb_annex_uncompressed"
+    "test_typeweb_export"
+    "test_submodule_url"
+    "test_uncurl_progress_reporting_to_annex"
+    "test_archivist_retrieval"
+    "test_archivist_retrieval_legacy"
+
+    # hardcoded /bin path
+    "test_auto_if_wanted_data_transfer_path_restriction"
```

Comment:
This test is not in the code🤔

Reply:
I believe `datalad-next` uses some fancy form of test code generation. I copied the failing tests from the terminal and looked through their error messages. So, yes, this test did fail. I tried your suggestion of patching `/bin` but it actually isn't found (easily) in the code.